src/shared/sleep-config.c

```diff
@@ -102,6 +105,183 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
+        int discharge_rate, r;
+
+        filepath = "/var/lib/systemd/sleep/battery_discharge_percentage_rate_per_hour";
+
+        r = read_one_line_file(filepath, &hash_id_discharge_rate);
+        log_debug("Stored hash_id and discharge_rate is %s", hash_id_discharge_rate); /* remove */
+        if (r < 0)
+               return log_debug_errno(r, "Failed to read discharge rate from %s: %m", filepath);
+
+        r = get_system_battery_identifier_hash(current_hash_id);
+        if (r < 0)
+               return log_debug_errno(r, "Failed to generate system battery identifier hash: %m");
+
+        p = hash_id_discharge_rate;
+        r = extract_many_words(&p, " ", 0, &stored_hash_id, &stored_discharge_rate, NULL);
+        log_debug("stored hash_id is %s and stored discharge_rate is %s", stored_hash_id, stored_discharge_rate); /* remove */
+        if (r < 0)
+               return log_debug_errno(r, "Failed to parse hash_id and discharge_rate read from %s location: %m", filepath);
+
+        h = hexmem(current_hash_id, sizeof(current_hash_id));
+        if (!h)
+               return -ENOMEM;
+
+        if (!streq(h, stored_hash_id))
+               return log_debug_errno(SYNTHETIC_ERRNO(ENOENT), "Current dentifier do not match stored identifier: %m");
+        log_debug("hash id match"); /* remove */
+
+        r = safe_atoi(stored_discharge_rate, &discharge_rate);
+        if (r < 0)
+               return log_debug_errno(r, "Failed to parse discharge rate read from %s location: %m", filepath);
+
+        if (!battery_discharge_rate_is_valid(discharge_rate))
+               return log_debug_errno(SYNTHETIC_ERRNO(ERANGE), "Invalid battery discharge percentage rate per hour: %m");
+
+        return discharge_rate;
+}
+
+/* Write battery percentage discharge rate per hour along with system and battery identifier hash to file */
+int put_battery_discharge_rate(int estimated_battery_discharge_rate) {
+        _cleanup_free_ char *h = NULL;
+        uint8_t system_hash_id[SHA256_DIGEST_SIZE];
+        const char *filepath;
+        int r = 0;
+
+        if (!battery_discharge_rate_is_valid(estimated_battery_discharge_rate))
+               return log_debug_errno(SYNTHETIC_ERRNO(ERANGE), "Invalid battery discharge percentage rate per hour: %m");
+
+        r = get_system_battery_identifier_hash(system_hash_id);
+        if (r < 0)
+               return log_debug_errno(r, "Failed to generate system battery identifier hash: %m");
+
+        h = hexmem(system_hash_id, sizeof(system_hash_id));
+        if (!h)
+               return -ENOMEM;
+        log_debug("system_hash_id is %s and discharge_rate is %d", h, estimated_battery_discharge_rate); /* remove */
+        filepath = "/var/lib/systemd/sleep/battery_discharge_percentage_rate_per_hour";
+
+        r = write_string_filef(filepath,
+                        WRITE_STRING_FILE_CREATE|WRITE_STRING_FILE_MKDIR_0755,
+                        "%s %d", h, estimated_battery_discharge_rate);
```

Comment:
If you use 16ch indentation, then please add line break before `filepath`.

BTW, please use `#define` for the path, instead of `const char*`.